sharpd/sharp_vty.c

```diff
@@ -527,6 +527,46 @@ DEFPY (logpump,
 	return CMD_SUCCESS;
 }
 
+DEFPY (send_opaque,
+       send_opaque_cmd,
+       "sharp send opaque type (1-255) (1-1000)$count",
+       "Sharp Routing Protocol\n"
+       "Send messages for testing\n"
+       "Send opaque messages\n"
+       "Type code to send\n"
+       "Type code to send\n"
+       "Number of messages to send\n")
+{
+	sharp_opaque_send(type, count);
+	return CMD_SUCCESS;
+}
+
+DEFPY (send_opaque_reg,
+       send_opaque_reg_cmd,
+       "sharp send opaque <reg$reg | unreg> \
+       " FRR_IP_REDIST_STR_ZEBRA "$proto_str \
+        [{instance (0-1000) | session (1-1000)}] type (1-1000)",
+       "Sharp Routing Protocol\n"
+       "Send messages for testing\n"
+       "Send opaque messages\n"
+       "Send opaque registration\n"
+       "Send opaque unregistration\n"
+       FRR_IP_REDIST_HELP_STR_ZEBRA
+       "Daemon instance\n"
+       "Daemon instance\n"
+       "Session ID\n"
+       "Session ID\n"
+       "Opaque sub-type code\n"
+       "Opaque sub-type code\n")
+{
+	int proto;
+
+	proto = proto_redistnum(AFI_IP, proto_str);
+
+	sharp_opaque_reg_send((reg != NULL), proto, instance, session, type);
+	return CMD_SUCCESS;
```

Comment:
IMHO, there is no need for the send to register itself as there is no control on zebra_opaque to verify that the send has been registered to OPAQUE message before. But, I understand that it is for testing.

Reply:
No, we need the complete tuple, to ensure we don't end up in the mess with 'instance' again. the receiver asks to be sent messages on a specific zclient/zapi session - it can use its default session, which may or may not have an 'instance' value, or it can use a different one if it prefers.

Reply:
OK. Understand.